R/py-classes.R

```diff
@@ -342,14 +350,51 @@ r_formals_to_py__signature__ <- function(fn) {
     parent_env = parent_env
   )))
 
-  # TODO: think about best practice for how to delay creating this
-  # if defined in a package.
-  py_cls <- r_to_py.R6ClassGenerator(r_cls, convert)
-  assign(classname, py_cls, envir = parent_env)
-  invisible(py_cls)
+
+  if (delay_load)
+    py_class <- delayed_r_to_py_R6ClassGenerator(r6_class, convert)
+  else
+    py_class <- r_to_py.R6ClassGenerator(r6_class, convert)
+
+  assign(classname, py_class, envir = parent_env)
+  invisible(py_class)
 }
 
 if (getRversion() < "4.0")
   activeBindingFunction <- function(nm, env) {
     as.list.environment(env, all.names = TRUE)[[nm]]
   }
+
+
+delayed_r_to_py_R6ClassGenerator <- function(r6_class, convert) {
+  force(r6_class)
+  force(convert)
+
+  py_object <- new.env(parent = emptyenv())
+  attr(py_object, "class") <- c("python.builtin.type", "python.builtin.object")
+
+  fn <- get("py_callable_as_function",
```

Comment:
Actually, to avoid using the non-exported function could you implement something like:

```
py_callable_as_function2 <- function(callable, convert) {
  
  force(callable)
  force(convert)
  
  function(...) {
    
    result <- py_call(callable, ...)
    
    if (convert)
      result <- py_to_r(result)
      
    if (is.null(result))
      invisible(result)
    else
      result
    
  }
  
}
```

It seems that `py_call` simply does: 

```
  ensure_python_initialized()
  dots <- py_resolve_dots(list(...))
  py_call_impl(x, dots$args, dots$keywords)
}
```

Which seems the same as `py_callable_as_function` is doing internally:

```
dots <- py_resolve_dots(list(...))
result <- py_call_impl(callable, dots$args, dots$keywords)
```

Unless checking for python is problematic here, but doesn't seem to be, because it will only be evaluated the first time it's used later.